### PR TITLE
RESTWS-582 enable editing concept names

### PR DESCRIPTION
**Pull request by PawelGutkowski**
@rkorytkowski, please check this.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@PawelGutkowski can you include some unit tests?

**Comment by PawelGutkowski** [comment]
@dkayiwa I just wanted to give @rkorytkowski some insight, because that's early draft, and I've forgot about them, here they are.

**Comment by rkorytkowski** [comment]
@PawelGutkowski, you found a very clean way to do it. I approve the change. We can merge it unless @dkayiwa or @djazayeri have objections.

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8Test.java [hunk 1]
```diff
@@ -72,6 +80,82 @@ public String getDisplayProperty() {
...
 	}
+	
+	
+	@Test
+	public void testSetNames() throws Exception {
+		Concept instance = new Concept();
+		
+		List<ConceptName> otherNames = new ArrayList<ConceptName>();
+    	ConceptName otherName = new ConceptName();
+    	otherName.setLocale(Locale.ENGLISH);
```
Comment (dkayiwa) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8Test.java]: Can you fix the indention here?

File: omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8Test.java [hunk 2]
```diff
@@ -13,9 +13,17 @@
  */
 package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
 
+import static org.junit.Assert.*;
```
Comment (dkayiwa) [inline omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8Test.java]: We generally discourage wild card imports https://wiki.openmrs.org/display/docs/Coding+Conventions

**Comment by PawelGutkowski** [comment]
@dkayiwa thanks for Your tips, I have fixed the issues.

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java [hunk 3]
```diff
@@ -258,15 +261,53 @@ public static void setFullySpecifiedName(Concept instance, String name) {
...
+	 * so this method just adds new and removes deleted names.
+	 * 
 	 * @param instance
-	 * @param names
+	 * @param newNames
 	 */
 	@PropertySetter("names")
-	public static void setNames(Concept instance, List<ConceptName> names) {
-		for(ConceptName name : names){
+	public static void setNames(Concept instance, List<ConceptName> newNames) {
```
Comment (wluyima) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java]: Are these necessarily new names?

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java [hunk 4]
```diff
@@ -258,15 +261,53 @@ public static void setFullySpecifiedName(Concept instance, String name) {
...
+	 * @should return true if list contains name with same properties
+	 * @should return false if there is no concept with same properties
+	 * 
+	 * @param names
+	 * @param searched
+	 * @return
+	 */
+	public static boolean checkIfNamesContainNameByProperties(Collection<ConceptName> names, ConceptName searched) {
+		for(ConceptName name : names){
+			if(Objects.equals(name.getUuid(), searched.getUuid())) return true;
```
Comment (wluyima) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java]: Our code convention is if clause statements should be inside curly braces and the statements should be on a new line

**Comment by PawelGutkowski** [comment]
@wluyima You're right, these are not necessarily new names, i have fixed argument name and if statement as well.

**Pull request merged**